docs/protocol/off-chain/app-sessions.mdx

```diff
@@ -0,0 +1,778 @@
+
+### Response
+
+| Parameter | Type | Description | Format/Structure | Example | Notes |
+|-----------|------|-------------|------------------|---------|-------|
+| `app_session_id` | string | Session identifier (echoed) | - | - | - |
+| `version` | number | Confirmed new version number | - | `2` | - |
+| `intent` | string | Confirmed intent (if applicable) | - | - | - |
+| `allocations` | Allocation[] | Confirmed final allocations | - | - | - |
+| `session_data` | string | Confirmed session data | - | - | - |
+| `locked_funds` | object | Updated locked funds per participant | - | `{"0xAlice": {"usdc": "75.0"}, "0xBob": {"usdc": "125.0"}}` | - |
+| `total_locked` | object | Total funds in session | - | `{"usdc": "200.0"}` | - |
+| `timestamp` | number | Unix timestamp of update | - | - | - |
+
+---
+
+## Intent System (NitroRPC/0.4)
+
+The intent system defines the type of operation being performed. Each intent has specific validation rules.
+
+### Intent: OPERATE (Redistribute Existing Funds)
+
+**Purpose**: Move funds between <Tooltip content="An entity (identified by a wallet address) that is part of a channel">participants</Tooltip> without changing total amount in session.
+
+**Rules**:
+- Sum of allocations MUST equal sum before operation
+- No funds added or removed from session
+- Quorum requirement MUST be met
+- Depositing <Tooltip content="An entity (identified by a wallet address) that is part of a channel">participant</Tooltip> signature NOT required
+
+**Example**:
+
+```
+Current state (version 1):
+  Alice: 100 USDC
+  Bob: 100 USDC
+  Total: 200 USDC
+
+Update (version 2, intent: "operate"):
+  Allocations: [
+    {"participant": "0xAlice", "asset": "usdc", "amount": "75.0"},
+    {"participant": "0xBob", "asset": "usdc", "amount": "125.0"}
+  ]
+  
+Result:
+  Alice: 75 USDC (-25)
+  Bob: 125 USDC (+25)
+  Total: 200 USDC (unchanged) ✓
+
+Validation: Sum before (200) == Sum after (200) ✓
+```
+
+**Use Cases**:
+- Record game outcome (winner gets opponent's stake)
+- Update prediction market positions
+- Rebalance shared pool
+- Penalize or reward <Tooltip content="An entity (identified by a wallet address) that is part of a channel">participants</Tooltip>
+
+:::tip OPERATE Intent
+Use OPERATE for simple fund redistributions within the session. The total amount remains constant—funds just move between <Tooltip content="An entity (identified by a wallet address) that is part of a channel">participants</Tooltip>.
+:::
+
+---
+
+### Intent: DEPOSIT (Add Funds to Session)
+
+**Purpose**: Add funds from a <Tooltip content="An entity (identified by a wallet address) that is part of a channel">participant's</Tooltip> <Tooltip content="An abstraction that aggregates a user's funds across multiple blockchain networks, managed by Clearnode">unified balance</Tooltip> into the session.
+
+**Rules**:
+- Sum of allocations MUST be greater than sum before operation
```

Comment:
I'd say that each allocation should be GREATER or EQUAL to the one before the operation